app/src/test/java/com/hedvig/app/feature/genericauth/OtpInputViewModelTest.kt

```diff
@@ -0,0 +1,211 @@
+package com.hedvig.app.feature.genericauth
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.hedvig.app.R
+import com.hedvig.app.authenticate.AuthenticationTokenService
+import com.hedvig.app.feature.genericauth.otpinput.OtpInputViewModel
+import com.hedvig.app.feature.genericauth.otpinput.OtpResult
+import com.hedvig.app.feature.genericauth.otpinput.ReSendOtpCodeUseCase
+import com.hedvig.app.feature.genericauth.otpinput.ResendOtpResult
+import com.hedvig.app.feature.genericauth.otpinput.SendOtpCodeUseCase
+import com.hedvig.app.util.coroutines.MainCoroutineRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class OtpInputViewModelTest {
+
+    private lateinit var viewModel: OtpInputViewModel
+    private var authToken: String? = "testToken"
+
+    @ExperimentalCoroutinesApi
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private var otpResult: OtpResult = OtpResult.Success("authtest")
+    private var resendOtpResult: ResendOtpResult = ResendOtpResult.Success("authtest")
+
+    @Before
+    fun setup() {
+        viewModel = OtpInputViewModel(
+            otpId = "testId",
+            credential = "test@email.com",
+            authenticationTokenService = object : AuthenticationTokenService {
+                override var authenticationToken: String?
+                    get() = authToken
+                    set(value) {
+                        authToken = value
+                    }
+            },
+            sendOtpCodeUseCase = object : SendOtpCodeUseCase {
+                override suspend fun invoke(otpId: String, otpCode: String): OtpResult {
+                    return otpResult
+                }
+            },
+            reSendOtpCodeUseCase = object : ReSendOtpCodeUseCase {
+                override suspend fun invoke(credential: String): ResendOtpResult {
+                    return resendOtpResult
+                }
+            }
+        )
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun testNetworkError() = runBlockingTest {
```

Comment:
Should these tests use the `runBlockingTest`-method from the `TestCoroutineDispatcher` in the `MainCoroutineRule`? 🤔